### PR TITLE
update backend module's transication calculation logical

### PR DESCRIPTION
**Pull request by louisliu2048**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/order/keeper/keeper.go [hunk 1]
```diff
@@ -323,6 +324,11 @@ func (k Keeper) GetUpdatedOrderIDs() []string {
 	return k.cache.getUpdatedOrderIDs()
 }
 
+// GetTxHandlerMsgResult: be careful, only call by backend module, other module should never use it!
+func (k Keeper) GetTxHandlerMsgResult() []bitset.BitSet {
+	return k.cache.toggleCopyTxHandlerMsgResult()
```
Comment (zhongqiuwood) [inline x/order/keeper/keeper.go:329]: add following code in GetTxHandlerMsgResult?
if !k.enableBackend {
   return nil
}

File: x/order/keeper/memory_cache.go [hunk 2]
```diff
@@ -110,6 +117,16 @@ func (c *Cache) getUpdatedOrderIDs() []string {
 	return c.updatedOrderIDs
 }
 
+// toggleCopyTxHandlerMsgResult: copy and reset the handlerTxMsgResult
+func (c *Cache) toggleCopyTxHandlerMsgResult() []bitset.BitSet {
+	txResultCopy := make([]bitset.BitSet, 0, len(c.handlerTxMsgResult))
+	txResultCopy = append(txResultCopy, c.handlerTxMsgResult...)
+	//copy(txResultCopy, c.handlerTxMsgResult)
```
Comment (zhongqiuwood) [inline x/order/keeper/memory_cache.go:124]: remove this line

File: x/order/handler_test.go [hunk 3]
```diff
@@ -436,7 +436,8 @@ func TestHandleMsgMultiNewOrder(t *testing.T) {
 	keeper.SetProductLock(ctx, types.TestTokenPair, &types.ProductLock{})
 	result1 := handler(ctx, msg)
 	res1 := parseOrderResult(result1)
-	require.EqualValues(t, sdk.CodeInternal, res1[0].Code)
+	require.Nil(t, res1)
+	//require.EqualValues(t, sdk.CodeInternal, res1[0].Code)
```
Comment (zhongqiuwood) [inline x/order/handler_test.go:440]: remove this line

File: x/order/handler_test.go [hunk 4]
```diff
@@ -325,8 +325,8 @@ func TestHandleMsgCancelOrderInvalid(t *testing.T) {
 	msg = types.NewMsgCancelOrder(addrKeysSlice[0].Address, order.OrderID)
 	result = handler(ctx, msg)
 	orderRes = parseOrderResult(result)
-	require.NotNil(t, orderRes)
-	require.EqualValues(t, sdk.CodeInternal, orderRes[0].Code)
+	require.Nil(t, orderRes)
+	//require.EqualValues(t, sdk.CodeInternal, orderRes[0].Code)
```
Comment (zhongqiuwood) [inline x/order/handler_test.go:329]: remove this line

File: x/order/handler_test.go [hunk 5]
```diff
@@ -284,21 +284,21 @@ func TestHandleMsgCancelOrderInvalid(t *testing.T) {
...
+	//require.EqualValues(t, sdk.CodeUnknownRequest, orderRes[0].Code)
 	// busy product
 	keeper.SetProductLock(ctx, order.Product, &types.ProductLock{})
 	msg = types.NewMsgCancelOrder(addrKeysSlice[0].Address, order.OrderID)
 	result = handler(ctx, msg)
 	orderRes = parseOrderResult(result)
-	require.NotNil(t, orderRes)
-	require.EqualValues(t, sdk.CodeInternal, orderRes[0].Code)
+	require.Nil(t, orderRes)
+	//require.EqualValues(t, sdk.CodeInternal, orderRes[0].Code)
```
Comment (zhongqiuwood) [inline x/order/handler_test.go:301]: remove this line

File: x/order/handler_test.go [hunk 6]
```diff
@@ -108,8 +108,8 @@ func TestHandleMsgNewOrderInvalid(t *testing.T) {
 	msg := types.NewMsgNewOrder(addrKeysSlice[0].Address, "nobb_"+common.NativeToken, types.BuyOrder, "10.0", "1.0")
 	result := handler(ctx, msg)
 	orderRes := parseOrderResult(result)
-	require.NotNil(t, orderRes)
-	require.EqualValues(t, sdk.CodeUnknownRequest, orderRes[0].Code)
+	require.Nil(t, orderRes)
+	//require.EqualValues(t, sdk.CodeUnknownRequest, orderRes[0].Code)
```
Comment (zhongqiuwood) [inline x/order/handler_test.go:112]: remove commented out code

File: x/order/keeper/keeper.go (hunk 1 above)
Comment (louisliu2048) [reply x/order/keeper/keeper.go:332]: Yes, it will be more clearly to user who want to call this function.

**Pull request merged**